### PR TITLE
Add database workflow and integration test for it.

### PR DESCRIPTION
**Pull request by wangxf123456**

### CODE REVIEW

File: django_cloud_deploy/integration_tests/cloudlib_test.py [hunk 1]
```diff
@@ -1,88 +0,0 @@
-# Copyright 2018 Google LLC
```
Comment (wangxf123456) [inline django_cloud_deploy/integration_tests/cloudlib_test.py]: I added a new integration test for create_and_setup database workflow, which includes testing of cloud sql instance creation and database creation.

File: django_cloud_deploy/workflow/_database.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+from typing import Callable
+
+from django_cloud_deploy.cloudlib import database
+
+from google.auth import credentials
+
+
+class DatabaseWorkflow(object):
+    """A class to control workflow for setting up database for Django app."""
```
Comment (brianquinlan) [inline django_cloud_deploy/workflow/_database.py]: the
```suggestion
    """A class to control workflow for setting up the database for a Django app."""
```

File: django_cloud_deploy/cloudlib/database.py [hunk 3]
```diff
@@ -168,3 +173,114 @@ def set_database_password(self, project_id: str, instance: str, user: str,
...
+                project_id, region, instance_name)
+            instance_flag = '-instances={}=tcp:{}'.format(
+                instance_connection_string, port)
+            process = pexpect.spawn(
+                cloud_sql_proxy_path, args=[instance_flag])
+            # Make sure cloud sql proxy is started before doing the real work
+            process.expect('Ready for new connections', timeout=5)
+            yield
+        except pexpect.exceptions.TIMEOUT:
+            raise DatabaseError(
```
[the remark this answers is not in the document]
Comment (wangxf123456) [reply django_cloud_deploy/cloudlib/database.py]: Done.

**Comment by enriquejosepadilla** [comment]
LGTM, nice refactor :+1:

**Pull request merged**